src/vertex_pulling/extract.rs

```diff
@@ -38,15 +37,8 @@ pub(crate) fn extract_cuboids(
     let materials_indices = materials.write_uniforms(&mut materials_uniforms);
 
     let mut extracted_entities = Vec::with_capacity(*prev_extracted_entities_size);
-    for (
-        entity,
-        cuboids,
-        transform,
-        materials_id,
-        maybe_visibility,
-        instance_buffer_needs_update,
-    ) in cuboids.iter()
-    {
+    for (entity, cuboids, transform, materials_id, maybe_visibility) in &cuboids {
+        let instance_buffer_needs_update = cuboids.is_added() || cuboids.is_changed();
```

Comment:
Following the docs there's an [`is_added()` etc now](https://docs.rs/bevy_ecs/latest/bevy_ecs/query/struct.Added.html), as I no longer seem to be able to use the `Added`/`Changed` traits to get a boolean value out of the query. Something about `ReadOnlyQueryData` not being implemented, and I'm not familiar enough with `bevy` internals to debug that further.